catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java

```diff
@@ -164,6 +177,46 @@ public Metacard getMetacard(String id)
         return result.getMetacard();
     }
 
+    /**
+     * Returns all associated metacards given two relatable attributes.
+     *
+     * @param primaryFilterAttribute   The primary attribute to search against
+     * @param secondaryFilterAttribute The secondary attribute to search against
+     * @param primaryAttributeValue    A primary attribute value
+     * @param secondaryAttributeValue  A secondary attribute value
+     * @return A set of metacards, filtered by the given attributes and it's values
+     * @throws UnsupportedQueryException  Error
+     * @throws SourceUnavailableException Error
+     * @throws FederationException        Error
+     */
+    public List<Metacard> getAssociatedMetacardsByTwoAttributes(String primaryFilterAttribute,
+            String secondaryFilterAttribute, String primaryAttributeValue,
+            String secondaryAttributeValue)
+            throws UnsupportedQueryException, SourceUnavailableException, FederationException {
+        LinkedList<Metacard> resultList = new LinkedList<>();
```

Comment:
❓ Is there a specific reason for choosing a LinkedList here?

Reply:
☝️ same question

Reply:
no other reason than this is how the original author wrote the code. But it looks like I can clean it up a bit